addons/product/models/product_template.py

```diff
@@ -598,19 +598,21 @@ def _create_variant_ids(self):
                 # For each possible variant, create if it doesn't exist yet.
                 for combination_tuple in all_combinations:
                     combination = self.env['product.template.attribute.value'].concat(*combination_tuple)
-                    if combination in existing_variants:
-                        current_variants_to_activate += existing_variants[combination]
-                    else:
-                        current_variants_to_create.append({
-                            'product_tmpl_id': tmpl_id.id,
-                            'product_template_attribute_value_ids': [(6, 0, combination.ids)],
-                            'active': tmpl_id.active,
-                        })
-                        if len(current_variants_to_create) > 1000:
-                            raise UserError(_(
-                                'The number of variants to generate is too high. '
-                                'You should either not generate variants for each combination or generate them on demand from the sales order. '
-                                'To do so, open the form view of attributes and change the mode of *Create Variants*.'))
+                    is_combination_possible = tmpl_id._is_combination_possible_by_config(combination, ignore_no_variant=True)
+                    if is_combination_possible:
```

Comment:
maybe use "if not ... continue" to avoid diff in the rest and crazy indent levels

Reply:
Ok